### DISCUSSION TITLE
Use of print() over python's logging and warning system

### DISCUSSION THREAD

**Discussion by ligerlac** [fastmachinelearning/hls4ml#1071] [General]
Hello everyone,
`hls4ml` often uses `print()` over python's `warnings` and `logging` modules. For example, in `hls4ml/converters/__init__.py`:
```
try:
    lib = importlib.import_module(__name__ + f'.{model_type}.' + module[:-3])
    ...
except ImportError as err:
    print(f'WARNING: Failed to import handlers from {module}: {err.msg}.')
```
we only use the `keras` converter in our project and would like to suppress the warning that `torch` is not installed. However, this is quite cumbersome as it is a `print('WARNING: ...')` statement, not an actual python warning.

I can create a PR to transform these makeshift warnings into actual warnings. But I first want to discuss here if there is maybe a good reason not to do that.

Cheers

**Comment by vloncar** [comment]
We discussed this and will do it after the 1.0 release. We'll change all print statements and replace some with `warnings.warn()` and some with `logging.warn()`. Also some form of debugging level of logging will be introduced to track the transformations

**Reply by ligerlac** [reply]
Okay, sounds good. Let me know if I can help you with that.

**Reply by vloncar** [reply]
Sure, I'll let you know. If you're keen on doing some maintenance work in the meantime we can figure something else out :-)
